src/annotator/notebook.js

```diff
@@ -1,147 +1,30 @@
-    this._prevGroupId = null;
-
     /**
      * Un-styled shadow host for the notebook content.
-     *
      * This isolates the notebook from the page's styles.
      */
     this._outerContainer = document.createElement('hypothesis-notebook');
-    this.element.appendChild(this._outerContainer);
-
-    /**
-     * Lazily-initialized container for the notebook iframe. This is only created
-     * when the notebook is actually used.
-     *
-     * @type {HTMLElement|null}
-     */
-    this.container = null;
-
-    this._emitter.subscribe('openNotebook', groupId => {
-      this._groupId = groupId;
-      this.open();
-    });
-  }
-
-  _update() {
-    const container = this._initContainer();
-
-    // Create a new iFrame if we don't have one at all yet, or if the
-    // groupId has changed since last use
-    const needIframe =
-      !this.frame || !this._prevGroupId || this._prevGroupId !== this._groupId;
-    this._prevGroupId = this._groupId;
-
-    if (needIframe) {
-      this.frame?.remove();
-      this.frame = createNotebookFrame(this.options, this._groupId);
-      container.appendChild(this.frame);
-    }
-  }
-
-  open() {
-    const container = this._initContainer();
-    this._update();
-    container.classList.add('is-open');
-    container.style.display = '';
-
-    // The overflow CSS property is set to hidden to prevent scrolling of the guest page,
-    // while the notebook is shown as modal. It is restored on the close method.
-    // I believe this hack only works if this.element points to document.body of the guest page.
-    this.originalOverflowStyle = this.element.style.overflow;
-    this.element.style.overflow = 'hidden';
-  }
-
-  close() {
-    if (this.container) {
-      this.container.classList.remove('is-open');
-      this.container.style.display = 'none';
-    }
-    this.element.style.overflow = /** @type {string} */ (this
-      .originalOverflowStyle);
-  }
-
-  destroy() {
-    this._outerContainer.remove();
-    this._emitter.destroy();
-  }
-
-  _initContainer() {
-    if (this.container) {
-      return this.container;
-    }
-
+    element.appendChild(this._outerContainer);
     const shadowRoot = createShadowRoot(this._outerContainer);
-    this.container = document.createElement('div');
-    this.container.style.display = 'none';
-    this.container.className = 'notebook-outer';
-    shadowRoot.appendChild(this.container);
-
-    const onClose = () => {
-      this.close();
-      this._emitter.publish('closeNotebook');
-    };
 
     render(
-      <div className="Notebook__controller-bar">
-        <Button
-          icon="cancel"
-          className="Notebook__close-button"
-          buttonText="Close"
-          title="Close the Notebook"
-          onClick={onClose}
-        />
-      </div>,
-      this.container
+      <NotebookContainer eventBus={eventBus} config={config} />,
+      shadowRoot
     );
+  }
 
-    return this.container;
+  destroy() {
+    this._outerContainer.remove();
```

Comment:
This will work as long as there are no effects or state updates pending in the `NotebookContainer`. In Preact the easiest way to flush effects that I'm aware of is to do `render(null, container)`. There isn't an `unmountComponentAtNode` function that React has. I think we make this mistake in several other components, but not running cleanup here has more significant effects (eg. not undoing the `<body>` style changes).